### PR TITLE
refactor: move to lerna monorepo

### PR DESCRIPTION
**Pull request by Rubilmax**
We are required to move to a monorepo to be able to port the ethers encoding SDK to viem and maintain both at the same time

Hardhat's compilation in packages is hacky: it uses symbolic links of root `src` to local `contracts`. But it works!

### CODE REVIEW

File: biome.json [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    "ignore": [
+      "out",
+      "cache",
+      "artifacts",
+      "lib",
+      "dist",
+      ".pnp.*",
+      ".vscode",
+      "package.json",
+      "packages/ethers/src/types"
```
Comment (julien-devatom) [inline biome.json:13]: you can also use `VcsConfiguration.useIgnoreFile: true`, but it wont include some artifacts (only `.gitignore` ones)

**Review by julien-devatom — APPROVED** [review]
config is only used for testing right? Wdyt about naming it "test-config"? Find it cleaner

File: tsconfig.json [hunk 2]
```diff
@@ -1,15 +1,15 @@
 {
```
Comment (adhusson) [inline tsconfig.json:1]: What is the purpose of a tsconfig at the toplevel?

File: hardhat.config.ts [hunk 3]
```diff
@@ -1,18 +1,5 @@
...
-import "tsconfig-paths/register";
-
-import "@nomicfoundation/hardhat-chai-matchers";
-import "@nomicfoundation/hardhat-ethers";
-import "@nomicfoundation/hardhat-foundry";
-import "@nomicfoundation/hardhat-network-helpers";
-import "@typechain/hardhat";
-
-dotenv.config();
+import "dotenv/config";
```
Comment (adhusson) [inline hardhat.config.ts:3]: What is the purpose of a hardhat config at the toplevel?

File: README.md [hunk 4]
```diff
@@ -6,11 +6,11 @@
 
 ![bundler-3](https://github.com/morpho-org/morpho-blue-bundlers/assets/74971347/827630e1-7abc-4f9d-a494-fe3fa7aa7053)
 
-Each Bundler is a domain-specific abstract layer of contract that implements some functions that can be bundled in a single call by EOAs to a single contract. They all inherit from [`BaseBundler`](./src/BaseBundler.sol) that enables bundling multiple function calls into a single `multicall(bytes[] calldata data)` call to the end bundler contract. Each chain-specific bundler is available under their chain-specific folder (e.g. [`ethereum`](./src/ethereum/)).
+Each Bundler is a domain-specific abstract layer of contract that implements some functions that can be bundled in a single call by EOAs to a single contract. They all inherit from [`BaseBundler`](./contracts/BaseBundler.sol) that enables bundling multiple function calls into a single `multicall(bytes[] calldata data)` call to the end bundler contract. Each chain-specific bundler is available under their chain-specific folder (e.g. [`ethereum`](./contracts/ethereum/)).
```
[the remark this answers is not in the document]
Comment (adhusson) [reply README.md:9]: This may be intended to be put in `./packages/ethers`.

[the remark this answers is not in the document]
Comment (Rubilmax) [reply README.md:9]: Thanks! It's an artifact of all the things I tried, I'll revert

File: package.json [hunk 5]
```diff
@@ -16,78 +16,47 @@
...
+  "packageManager": "yarn@4.4.0",
+  "workspaces": [
+    "packages/*"
+  ],
   "scripts": {
-    "prepare": "husky install && forge install",
-    "build:pkg": "yarn build:hardhat --force && tsc --build ./tsconfig.build.json",
-    "build:forge": "FOUNDRY_PROFILE=build forge build",
-    "build:hardhat": "hardhat compile",
+    "prepare": "husky && forge install",
```
Comment (adhusson) [inline package.json:25]:
```suggestion
    "prepare": "husky install && forge install",
```

File: biome.json (hunk 1 above)
Comment (Rubilmax) [reply biome.json:13]: I think it's usually not what we want: we may want to format files not committed and vice versa, not format some files that are committed (see here with types generated directory)

File: package.json [hunk 6]
```diff
@@ -16,78 +16,47 @@
...
+    "build": "FOUNDRY_PROFILE=build forge build",
     "typecheck": "tsc --noEmit",
```
Comment (adhusson) [inline package.json:27]: `yarn build` fails with a lot of `error TS2307: Cannot find module`

File: packages/ethers/package.json [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+    "url": "https://github.com/morpho-labs"
+  },
+  "bugs": {
+    "url": "https://github.com/morpho-labs/morpho-blue-bundlers/issues"
+  },
+  "homepage": "https://github.com/morpho-labs/morpho-blue-bundlers#readme",
+  "scripts": {
+    "prepublish": "yarn build",
+    "prepare": "ln -sfn ../../src/ ./contracts && ln -sfn ../../lib/ ./lib",
+    "typecheck": "tsc --noEmit",
```
Comment (adhusson) [inline packages/ethers/package.json:22]: yarn build fails with a lot of `error TS2307: Cannot find module`

File: tsconfig.json (hunk 2 above)
Comment (Rubilmax) [reply tsconfig.json:1]: To be able to share base configuration across ts packages

File: hardhat.config.ts (hunk 3 above)
Comment (Rubilmax) [reply hardhat.config.ts:3]: To share (inherit from) base configuration across hardhat packages

File: package.json (hunk 5 above)
Comment (Rubilmax) [reply package.json:25]: Install command is deprecated in v9

File: package.json (hunk 6 above)
Comment (Rubilmax) [reply package.json]: Removed

File: packages/ethers/package.json (hunk 7 above)
Comment (adhusson) [reply packages/ethers/package.json:22]: ok gone since update

**Review by adhusson — DISMISSED** [review]

**Review by julien-devatom — APPROVED** [review]

**Review by adhusson — APPROVED** [review]

**Pull request closed without merging**